orchestrator/careplanservice/service.go

```diff
@@ -674,3 +675,63 @@ func (s *Service) checkAllowUnmanagedOperations() error {
 	}
 	return nil
 }
+
+// validateLiteralReferences validates the literal references in the given resource.
+// Literal references may be an external URL, but they MUST use HTTPS and be a child of a FHIR base URL
+// registered in the CSD. This prevents unsafe external references (e.g. accidentally exchanging resources over HTTP),
+// and gives more confidence that the resource can safely be fetched by SCP-nodes.
+func (s *Service) validateLiteralReferences(ctx context.Context, resource any) error {
+	// Literal references are "reference" fields that contain a string. This can be anywhere in the resource,
+	// so we need to recursively search for them.
+	resourceAsJson, err := json.Marshal(resource)
+	if err != nil {
+		// would be very weird
+		return err
+	}
+	resourceAsMap := make(map[string]interface{})
+	err = json.Unmarshal(resourceAsJson, &resourceAsMap)
+	if err != nil {
+		// would be very weird
+		return err
+	}
+
+	literalReferences := make(map[string]string)
+	collectLiteralReferences(resourceAsMap, []string{}, literalReferences)
+	for path, reference := range literalReferences {
+		lowerCaseRef := strings.ToLower(reference)
+		if strings.HasPrefix(lowerCaseRef, "http://") {
```

Comment:
Seems strange to check twice like this, why not something like:

```
if !strings.HasPrefix(lowerCaseRef, "https://") {
			return coolfhir.BadRequest(fmt.Sprintf("literal reference is URL, only scheme https:// is allowed (path=%s)",  path))
		}
```

Reply:
Since it doesn't need to be a URL, it can also be a local reference (e.g.`Patient/123`), or `urn:uid:1234`

Reply:
Ahh, fair point